### PR TITLE
Flexible PAT Rebase on Correct-Util-Scale

### PR DESCRIPTION
**Pull request by danielsclint**
This not ready to be merged, but I needed something to create a discussion around.

Everything mechanically works, and most of the tests work. I had to make a couple of changes to the C++ that need further review.

### CODE REVIEW

File: src/pathfinder.cpp [hunk 1]
```diff
@@ -855,7 +855,7 @@ namespace fasttrips {
         NamedWeights::const_iterator ivt_weight = weights.find(ivt_str);
         if ((fare_attr != attributes.end()) && (ivt_weight != weights.end())) {
             //       (60 min/hour)*(hours/vot currency)*(ivt_weight) x (currency)
-            cost += (60.0/path_spec.value_of_time_) * ivt_weight->second * fare_attr->second;
+            cost += (60.0/path_spec.value_of_time_) * ivt_weight->second.weight_ * fare_attr->second;
```
Comment (danielsclint) [inline src/pathfinder.cpp]: When @lmz made the changes to support the different growth curves in C++, she changed NamedWeights from a simple dictionary to a more complex structure. I put in this change and the one below to get everything to compile. This needs and extra set of eyes from @lmz or @e-lo.

Comment (lmz) [reply src/pathfinder.cpp]: I looked at this but I'm not sure; this bit of code was introduced by @bhargavasana related to the utils (https://github.com/BayAreaMetro/fast-trips/commit/4410852eb4d4ad2b7b1888e8637ead54aa930899)

**Comment by danielsclint** [comment]
The problem is in the Python code. Pathset.calculate_overlap() method. If the overalap is set to anything besides 'None' the code gets stuck in a loop. I'm investigating, and I should have a patch later today.

**Comment by danielsclint** [comment]
The problem was in the configuration. The default chunk size for calculating overlap is way too big for my laptop. Once I specified an explicit chunk size in the configuration file, it was good to go. Let's make sure all the tests on Travis go okay, but I think this one may finally be good to go.

**Comment by e-lo** [comment]
@danielsclint : Looks like they failed b/c of a timeout on Travis.  Endless loop or Travis timeout?  Do they run locally?

**Comment by danielsclint** [comment]
I think so. I just kicked them off locally again. I’ll check back in a few hours.

Clint
________________________________

File: src/pathfinder.cpp (hunk 1 above)
Comment (bhargavasana) [reply src/pathfinder.cpp]: Hi Clint, the changes you made look good to me!

**Comment by e-lo** [comment]
@danielsclint confirmed that all the tests run fine locally on his machine.

**Pull request merged**